src/pwa-buildpack/project-setup/create-configuration-files/index.md

```diff
@@ -0,0 +1,249 @@
+        entry: {
+            client: path.resolve(themePaths.src, 'index.js')
+        },
+        output: {
+            path: themePaths.output,
+            publicPath: process.env.MAGENTO_BACKEND_PUBLIC_PATH,
+            filename: '[name].js',
+            chunkFilename: '[name].js'
+        },
+        module: {
+            rules: [
+                {
+                    include: [themePaths.src],
+                    test: /\.js$/,
+                    use: [
+                        {
+                            loader: 'babel-loader',
+                            options: { cacheDirectory: true }
+                        }
+                    ]
+                },
+                {
+                     test: /\.css$/,
+                     use: [
+                         'style-loader',
+                         {
+                             loader: 'css-loader',
+                             options: {
+                                 importLoaders: 1
+                             }
+                         }
+                     ]
+                 }
+             ]
+         },
+         resolve: await MagentoResolver.configure({
+             paths: {
+                 root: __dirname
+             }
+         }),
+         plugins: [
+             new MagentoRootComponentsPlugin(),
+             new webpack.NoEmitOnErrorsPlugin(),
+             new webpack.EnvironmentPlugin({
+                 NODE_ENV: env.mode,
+                 SERVICE_WORKER_FILE_NAME: 'sw.js'
+             })
+         ]
+
+    };
+
+    return config;
+}
+```
+
+This configuration sets up Webpack for your development environment.
+Some important things to note in this configuration:
+
+* How it configures the `MagentoResolver` service
+* The inclusion of the `MagentoRootComponent` as a plugin
+* The use of `webpack.EnvironmentPlugin` to pass environment variables
+
+### Add development mode configuration
+
+Add the following development mode configuration before returning the `config` object:
+
+``` javascript
+if (env.mode === "development") {
+    config.devServer = await PWADevServer.configure({
+        publicPath: process.env.MAGENTO_BACKEND_PUBLIC_PATH,
+        backendDomain: process.env.MAGENTO_BACKEND_DOMAIN,
+        serviceWorkerFileName: process.env.SERVICE_WORKER_FILE_NAME,
+        paths: themePaths,
+        id: 'magento-my-theme'
+    });
+
+    // A DevServer generates its own unique output path at startup. It needs
+    // to assign the main outputPath to this value as well.
+
+    config.output.publicPath = config.devServer.publicPath;
+
+    config.plugins.push(
+         new ServiceWorkerPlugin({
+             env,
+             paths: themePaths,
+             enableServiceWorkerDebugging: false,
+             serviceWorkerFileName: process.env.SERVICE_WORKER_FILE_NAME
+         })
+     );
+
+     config.plugins.push(
+         new webpack.HotModuleReplacementPlugin()
+     );
+} else if (env.mode === "production") {
```

Comment:
Maybe we should change this so that it dies under any other string but `development`. Like:

```js
} else {
  throw Error('Only "development" mode is currently supported. Please pass "--env.mode development" on the command line.');
}
```